src/inert.js

```diff
@@ -18,638 +18,322 @@
+      } else {
+        this._rootElement.removeAttribute('tabindex');
+      }
 
-      // Use `delete` to restore native focus method.
-      if (this._overrodeFocusMethod)
-        delete this._node.focus;
+      this._observer = null;
+      this._rootElement = null;
+      this._rootTabindex = null;
     }
-    this._node = null;
-    this._inertRoots = null;
-
-    this._destroyed = true;
   }
 
   /**
-   * @type {boolean} Whether this object is obsolete because the managed node is no longer inert.
-   * If the object has been destroyed, any attempt to access it will cause an exception.
+   * InertManager is a per-document singleton object which manages all inert roots and nodes.
+   *
+   * When an element becomes an inert root by having an `inert` attribute set and/or its `inert`
+   * property set to `true`, the `setInert` method creates an `InertRoot` object for the element.
+   * The `InertRoot` in turn registers itself as managing all of the element's focusable descendant
+   * nodes via the `register()` method. The `InertManager` ensures that a single `InertNode`
+   * instance is created for each such node, via the `_managedNodes` map.
    */
-  get destroyed() {
-    return this._destroyed;
-  }
-
-  /**
-   * Throw if user tries to access destroyed InertNode.
-   */
-  _throwIfDestroyed() {
-    if (this.destroyed)
-      throw new Error('Trying to access destroyed InertNode');
-  }
-
-  /** @return {boolean} */
-  get hasSavedTabIndex() {
-    return '_savedTabIndex' in this;
-  }
-
-  /** @return {Node} */
-  get node() {
-    this._throwIfDestroyed();
-    return this._node;
-  }
-
-  /** @param {number} tabIndex */
-  set savedTabIndex(tabIndex) {
-    this._throwIfDestroyed();
-    this._savedTabIndex = tabIndex;
-  }
-
-  /** @return {number} */
-  get savedTabIndex() {
-    this._throwIfDestroyed();
-    return this._savedTabIndex;
-  }
-
-  /** Save the existing tabindex value and make the node untabbable and unfocusable */
-  ensureUntabbable() {
-    const node = this.node;
-    if (matches(node, _focusableElementsString)) {
-      if (node.tabIndex === -1 && this.hasSavedTabIndex)
-        return;
-
-      if (node.hasAttribute('tabindex'))
-        this._savedTabIndex = node.tabIndex;
-      node.setAttribute('tabindex', '-1');
-      if (node.nodeType === Node.ELEMENT_NODE) {
-        node.focus = function() {};
-        this._overrodeFocusMethod = true;
+  class InertManager {
+    /**
+     * @param {Document} document
+     */
+    constructor(document) {
+      if (!document)
+        throw new Error('Missing required argument; InertManager needs to wrap a document.');
+
+      /** @type {Document} */
+      this._document = document;
+
+      /**
+       * All inert roots known to this InertManager. In a map to allow looking up by Node.
+       * @type {Map<Node, InertRoot>}
+       */
+      this._inertRoots = new Map();
+
+      this.watchForInert = this._watchForInert.bind(this);
```

Comment:
Can we make this "private"? Maybe call it something like `this._boundWatchForInert`?